echopype/utils/data_proc_lvls.py

```diff
@@ -0,0 +1,66 @@
+"""
+Contains common functions used throughout the data
+processing levels.
+"""
+import pathlib
+from typing import Optional, Tuple, Union
+
+import xarray as xr
+
+from echopype.utils.io import check_file_existence, get_file_format, validate_output_path
+
+
+def validate_source_Sv(
+    source_Sv: Union[xr.Dataset, str, pathlib.Path], storage_options: Optional[dict]
+) -> Tuple[Union[xr.Dataset, str, xr.DataArray], Optional[str]]:
+    """
+    This function ensures that ``source_Sv`` is of the correct
+    type and validates the path of ``source_Sv``, if it is provided.
+
+    Parameters
+    ----------
+    source_Sv: xr.Dataset or str or pathlib.Path
+        If a Dataset this value contains the Sv data,
+        else it specifies the path to a zarr or netcdf file
+    storage_options: dict, optional
+        Any additional parameters for the storage backend, corresponding to the
+        path provided for ``source_Sv``
+
+    Returns
+    -------
+    source_Sv: xr.Dataset or str
+        A Dataset that contains the Sv data (which will be the same as
+        the input) or a validated path to a zarr or netcdf file
+    file_type: {"netcdf4", "zarr"}, optional
+        If ``source_Sv`` is a path then corresponds to the file type of input, else
+        is ``None``
```

Comment:
```suggestion
    file_type: {"netcdf4", "zarr"}, optional
        The file type of the input path if ``source_Sv`` is a path, otherwise ``None``
```